### ISSUE TITLE
Neural Networks for Collaborative Filtering

### ISSUE THREAD

**Issue by guevara** [guevara/read-it-later#4087]
Neural Networks for Collaborative Filtering<br>
<br>
http://bit.ly/2ls2gkh<br>
<br>
About 0 Minutes<br>
<br>
<div>
<div>
<div>
<div>
<div>
<pre>
__________________________________________________________________________________________________
Layer (type)                    Output Shape         Param #     Connected to                     
==================================================================================================
Item (InputLayer)               (None, 1)            0                                            
__________________________________________________________________________________________________
User (InputLayer)               (None, 1)            0                                            
__________________________________________________________________________________________________
Movie-Embedding-MLP (Embedding) (None, 1, 10)        16830       Item[0][0]                       
__________________________________________________________________________________________________
User-Embedding-MLP (Embedding)  (None, 1, 8)         7552        User[0][0]                       
__________________________________________________________________________________________________
FlattenMovies-MLP (Flatten)     (None, 10)           0           Movie-Embedding-MLP[0][0]        
__________________________________________________________________________________________________
FlattenUsers-MLP (Flatten)      (None, 8)            0           User-Embedding-MLP[0][0]         
__________________________________________________________________________________________________
dropout_1 (Dropout)             (None, 10)           0           FlattenMovies-MLP[0][0]          
__________________________________________________________________________________________________
dropout_3 (Dropout)             (None, 8)            0           FlattenUsers-MLP[0][0]           
__________________________________________________________________________________________________
Concat (Merge)                  (None, 18)           0           dropout_1[0][0]                  
                                                                 dropout_3[0][0]                  
__________________________________________________________________________________________________
dropout_5 (Dropout)             (None, 18)           0           Concat[0][0]                     
__________________________________________________________________________________________________
FullyConnected (Dense)          (None, 200)          3800        dropout_5[0][0]                  
__________________________________________________________________________________________________
Batch (BatchNormalization)      (None, 200)          800         FullyConnected[0][0]             
__________________________________________________________________________________________________
Dropout-1 (Dropout)             (None, 200)          0           Batch[0][0]                      
__________________________________________________________________________________________________
FullyConnected-1 (Dense)        (None, 100)          20100       Dropout-1[0][0]                  
__________________________________________________________________________________________________
Batch-2 (BatchNormalization)    (None, 100)          400         FullyConnected-1[0][0]           
__________________________________________________________________________________________________
Movie-Embedding-MF (Embedding)  (None, 1, 3)         5049        Item[0][0]                       
__________________________________________________________________________________________________
User-Embedding-MF (Embedding)   (None, 1, 3)         2832        User[0][0]                       
__________________________________________________________________________________________________
Dropout-2 (Dropout)             (None, 100)          0           Batch-2[0][0]                    
__________________________________________________________________________________________________
FlattenMovies-MF (Flatten)      (None, 3)            0           Movie-Embedding-MF[0][0]         
__________________________________________________________________________________________________
FlattenUsers-MF (Flatten)       (None, 3)            0           User-Embedding-MF[0][0]          
__________________________________________________________________________________________________
FullyConnected-2 (Dense)        (None, 50)           5050        Dropout-2[0][0]                  
__________________________________________________________________________________________________
dropout_2 (Dropout)             (None, 3)            0           FlattenMovies-MF[0][0]           
__________________________________________________________________________________________________
dropout_4 (Dropout)             (None, 3)            0           FlattenUsers-MF[0][0]            
__________________________________________________________________________________________________
FullyConnected-3 (Dense)        (None, 20)           1020        FullyConnected-2[0][0]           
__________________________________________________________________________________________________
Dot (Merge)                     (None, 1)            0           dropout_2[0][0]                  
                                                                 dropout_4[0][0]                  
__________________________________________________________________________________________________
Activation (Dense)              (None, 1)            21          FullyConnected-3[0][0]           
__________________________________________________________________________________________________
Concat-MF-MLP (Merge)           (None, 2)            0           Dot[0][0]                        
                                                                 Activation[0][0]                 
__________________________________________________________________________________________________
Combine-MF-MLP (Dense)          (None, 100)          300         Concat-MF-MLP[0][0]              
__________________________________________________________________________________________________
FullyConnected-4 (Dense)        (None, 100)          10100       Combine-MF-MLP[0][0]             
__________________________________________________________________________________________________
Prediction (Dense)              (None, 1)            101         FullyConnected-4[0][0]           
==================================================================================================
Total params: 73,955
Trainable params: 73,355
Non-trainable params: 600
__________________________________________________________________________________________________
</pre>
</div>
</div>
</div>
</div>
</div>
<br>
<br>
<br>
<br>
<br>
<br>
via nipunbatra.github.io http://bit.ly/2Eq3lnf<br>
<br>
May 23, 2019 at 02:38PM
